### ISSUE TITLE
DI cannot create BotFrameworkHttpAdapter (ambiguous constructor)

### ISSUE THREAD

**Issue by EricDahlvang** [microsoft/botbuilder-dotnet#1988]
## Version
master branch from repo

## Describe the bug
```cs
services.AddSingleton<IBotFrameworkHttpAdapter, BotFrameworkHttpAdapter>();
```
Results in: 
```cs
System.InvalidOperationException: Unable to activate type 'Microsoft.Bot.Builder.Integration.AspNet.Core.BotFrameworkHttpAdapter'. The following constructors are ambiguous:
Void .ctor(Microsoft.Bot.Connector.Authentication.ICredentialProvider, Microsoft.Bot.Connector.Authentication.IChannelProvider, Microsoft.Extensions.Logging.ILogger`1[Microsoft.Bot.Builder.Integration.AspNet.Core.BotFrameworkHttpAdapter])
Void .ctor(Microsoft.Extensions.Configuration.IConfiguration, Microsoft.Extensions.Logging.ILogger`1[Microsoft.Bot.Builder.Integration.AspNet.Core.BotFrameworkHttpAdapter])
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(Type serviceType, Type implementationType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetService(IServiceProvider sp, Type type, Type requiredBy, Boolean isDefaultParameterRequired)
   at lambda_method(Closure , IServiceProvider , Object[] )
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerActivatorProvider.<>c__DisplayClass4_0.<CreateActivator>b__0(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerFactoryProvider.<>c__DisplayClass5_0.<CreateControllerFactory>g__CreateController|0(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

## To Reproduce
1. Clone the repo
2. Change .TestBot so it is injecting BotFrameworkHttpAdapter instead of AdapterWithErrorHandler
3. Connect from the emulator

## Expected behavior
No error when injecting BotFrameworkHttpAdapter 

## Screenshots
n/a

## Additional context
This seems important.

[bug]

**Comment by EurekaChen** [comment]
how to resolve this problem?

**Comment by EricDahlvang** [comment]
@EurekaChen  Until a fix is committed, you can either comment out one of the constructors from BotFrameworkHttpAdapter, or specify one during di setup:

```cs
services.AddSingleton<IBotFrameworkHttpAdapter>(new BotFrameworkHttpAdapter(Configuration));
```

**Comment by johnataylor** [comment]
We think we can fix this and not loose the new scenario by making the new constructor protected.
 
The general (though not required) pattern is to subclass the BotFrameworkHttpAdapter. In your subclass you would set the error handler add middleware etc.

So the new constructor would only be available to those who are subclassing and that seems fine.

**Comment by johnataylor** [comment]
@EricDahlvang @EurekaChen See my previous comment. I think you should subclass.

**Comment by EurekaChen** [comment]
back to 4.4.3 can resolve it

**Comment by johnataylor** [comment]
we are fixing this in 4.5 (and patching in 4.4)

**Comment by johnataylor** [comment]
fixed in 4.4.5 and in master
